sparse_strips/vello_common/src/strip.rs

```diff
@@ -302,3 +303,188 @@ pub fn render(
         }
     }
 }
+
+/// Draw the strips of a rectangle. This is faster than using the normal path, because we
+/// do not need to go through the "flatten", "tiling" and "sort" stages, but can instead
+/// directly emit strips.
+///
+/// This method will clear the strip buffer initially.
+pub fn render_rect(
+    rect: &Rect,
+    strip_buf: &mut Vec<Strip>,
+    alpha_buf: &mut Vec<u8>,
+    width: u16,
+    height: u16,
+) {
+    // The idea for this fast path is as follows:
+    // - We generate strips of width 1 for the left as well as the right side of the rectangle. The
+    //   left side has a winding number of 0, the right side has a winding number of 1.
+    // - We generate a strip of the full rectangle width for the top and bottom part of the rectangle.
+    // - Of course, it's also possible that a rectangle has a width of less than 2 or a height of
+    //   less than 4. The current logic does account for those edge cases.
+    // - There could be some further optimizations (for example, if a rectangle is strip-aligned on
+    //   the y-axis, we don't need the strips for the top part of the rectangle), but I don't think
+    //   those edge cases are worth adding to the complexity of this method.
+
+    strip_buf.clear();
+
+    // Don't try to draw empty rectangles.
+    if rect.is_zero_area() {
+        return;
+    }
+
+    // Note that we currently deal with negative-area rects as positive-area rects.
+    // Shouldn't be a problem for solid fill, but might need some tweaking for gradient
+    // and pattern fills.
+    let (x0, x1, y0, y1) = (
+        rect.min_x().max(0.0) as f32,
+        rect.max_x().min(width as f64) as f32,
```

Comment:
for `min_x() == 30`, `max_x() == 40`, `width == 10`, this sets `x0 = 30` and `x1 = 10`. This propagates through to `x_start` and `x_end`. Later we check that `x_end - x_start >= 1`, but due to wraparound, that'll be true. The end result is that we write a lot of unnecessary alpha values.